### PR TITLE
Fix blank charts

### PR DESCRIPTION
**Pull request by terrysahaidak**
Fixes RNBW-1115

## What changed (plus any additional context for devs)

Sometimes opening charts would result in an empty chart. It can be related to some changes to the Reanimated 2 Scheduler in the latest 2.4.1 version. After some debugging, I noticed that we rerender charts many times during the first render
the log looks like this:
```
 set path undefined
 LOG  1
 LOG  1
 LOG  set path undefined
 LOG  set path undefined
 LOG  set path string
 LOG  set path undefined
 LOG  1
 LOG  set path undefined
 LOG  1
 LOG  set path undefined
 LOG  [1:26:09 пп] 😬 FallbackExplorer:: fetchOnchainBalances
 LOG  set path string
 ```
where 1 is useAnimatedStyle with chart opacity
and set path - is useAnimatedProps

So we render chart many times and some of the renders with an empty chart path, so we pass `d: ''` as an animated prop. So since we pass an empty string and then the actual path right away Reanimated doesn't schedule the second update so we show the empty chart.

In this fix, I made sure we don't rerender the chart if we pass an empty string to render. So it renders only once and rerenders the path only if it changes to a new path.

## Dev checklist for QA: what to test

- go to search
- navigate to charts on at least 10 tokens

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?

### CODE REVIEW

File: src/react-native-animated-charts/src/charts/linear/ChartPath.tsx [hunk 1]
```diff
@@ -106,297 +106,362 @@ function positionXWithMargin(x: number, margin: number, width: number) {
...
+  const translationX = useSharedValue<number | null>(null);
+  const translationY = useSharedValue<number | null>(null);
+
+  const setOriginData = useWorkletCallback((path: PathData, index?: number) => {
+    if (!path.data.length) {
+      return;
+    }
 
-    const resetGestureState = useWorkletCallback(() => {
+    if (typeof index === 'undefined') {
```
Comment (osdnk) [inline src/react-native-animated-charts/src/charts/linear/ChartPath.tsx:146]:
```suggestion
    if (index === undefined) {
```
I think we don't this patter at all

Comment (osdnk) [reply src/react-native-animated-charts/src/charts/linear/ChartPath.tsx:146]: but i see this is only whitespace change and we already discussed this and you had some strong opinions about this so w/e

File: src/react-native-animated-charts/src/charts/linear/ChartPath.tsx [hunk 2]
```diff
@@ -106,297 +106,362 @@ function positionXWithMargin(x: number, margin: number, width: number) {
...
-      <View style={{ height, width }}>
-        <LongPressGestureHandler
-          enabled={gestureEnabled}
-          maxDist={100000}
-          minDurationMs={0}
-          onGestureEvent={onGestureEvent}
-          shouldCancelWhenOutside={false}
-          {...longPressGestureHandlerProps}
+    };
+  });
```
Comment (osdnk) [inline src/react-native-animated-charts/src/charts/linear/ChartPath.tsx:364]: why this?

File: src/react-native-animated-charts/src/charts/linear/ChartPath.tsx [hunk 3]
```diff
@@ -106,297 +106,362 @@ function positionXWithMargin(x: number, margin: number, width: number) {
   }
 }
 
-export const ChartPath = React.memo(
```
Comment (osdnk) [inline src/react-native-animated-charts/src/charts/linear/ChartPath.tsx:109]: no longer memo?

File: src/react-native-animated-charts/src/charts/linear/ChartPath.tsx [hunk 4]
```diff
@@ -106,297 +106,362 @@ function positionXWithMargin(x: number, margin: number, width: number) {
...
-          originalY.value = '';
-          return;
-        }
-
-        originalX.value = path.data[index].x.toString();
-        originalY.value = path.data[index].y.toString();
-      },
-      []
-    );
+const ChartPathInner = ({
```
Comment (osdnk) [inline src/react-native-animated-charts/src/charts/linear/ChartPath.tsx:109]: no memo?

File: src/react-native-animated-charts/src/charts/linear/ChartPath.tsx (hunk 2 above)
Comment (osdnk) [reply src/react-native-animated-charts/src/charts/linear/ChartPath.tsx:364]: I mean why in `const pathAnimatedStyles` we sopped using an anonymous function

**Review by osdnk — APPROVED** [review]
I tested this on a few wallets and looks correct

**Review by tchayen — APPROVED** [review]
I checked on a couple wallets on both Android and iOS. No problems, no regressions. Looking good.

File: src/react-native-animated-charts/src/charts/linear/ChartPath.tsx (hunk 4 above)
Comment (terrysahaidak) [reply src/react-native-animated-charts/src/charts/linear/ChartPath.tsx:109]: There is one for the original ChartPath but yeah it won't hurt.

**Review by perunt — APPROVED** [review]
The same for me, I tried a few wallets on Android, and everything works like a charm

**Review by ibrahimtaveras00 — APPROVED** [review]
Went through assets on wallet view, discover tokens & pools, and search on both OS's, looks good 👍🏽

**Pull request merged**